### PR TITLE
feat(jumpstart): send jumpstart transactions

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR adds the plumbing for sending the transactions.
1. add a new redux state variable and actions to track the status of the transactions. there can only be one in flight at a time. after the transaction is completed, we should navigate the user to the share screen and clear the navigation stack. 
2. the restriction of one in-flight jumpstart tx at a time comes from the share link being available only during the send flow. so we should ensure that the status of the jumpstart transaction is reset when a new link is generated (this is done in the JumpstartEnterAmount screen)
3. prevent the user from sending a transaction -> navigating back to change the amount and submitting another transaction, by disabling the proceed button on the JumpstartEnterAmount screen if there is an in-flight transaction.
4. add saga to send the transactions. the user should remain on the confirmation screen until we have a result.
5. add redux migration

Not included in this PR:
1. error handling for when the transactions are not successful (waiting on design)
2. analytics
6. navigating to the next screen (placeholder for now)

### Test plan

https://github.com/valora-inc/wallet/assets/20150449/376df624-8ea8-49a3-aea5-9a3b2cd0eff2

### Related issues

- Related to RET-994

### Backwards compatibility

Y

### Network scalability

Y

### CODE REVIEW

**Review by jophish — COMMENTED** [review]
Left a couple of questions/comments!

File: src/jumpstart/saga.ts
Comment (kathaypacific) [reply src/jumpstart/saga.ts:227]: yes that's a great point...reading this function it does seem overly defensive against the serialized transactions despite that being something that we create ourselves in this flow. the intention of the explicit verification that the first tx is an approval tx isn't to decide whether or not to include the tx, rather to set an accurate value for the `approvedAmount` field of the pending approval transaction. 

after the pending transaction creators are created, they are passed along with the serialised transactions to `sendPreparedTransactions`, which will throw if these arrays have different lengths. 

i take your feedback about the readability of this function though, i think what might be less confusing is to throw early if the first tx is _not_ an approval...

File: src/jumpstart/JumpstartSendConfirmation.tsx
Comment (kathaypacific) [reply src/jumpstart/JumpstartSendConfirmation.tsx:42]: yes you're right, i didn't feel great about this implementation either. this is pretty analgous to clearing jest mocks in the `afterEach` vs `beforeEach` blocks, i do favour ensuring the state is clean at the start rather than the end of a flow (since we may never get to the end, as in your example). i'll tweak this and move the logic to the beginning of the flow

Comment (kathaypacific) [reply src/jumpstart/JumpstartSendConfirmation.tsx:43]: the changes i ended up making (also documented in the PR description) -
1. ensure the `depositStatus` is reset to `idle` when a new jumpstart link is created at the beginning of the flow. each link uniquely identifies each transaction.
2. block the "proceed" button on the enter amount screen if we can see there is an inflight or successful transaction already. this means the user needs to go back and launch the flow again, which is all good.

File: src/jumpstart/saga.ts
Comment (jeanregisser) [inline src/jumpstart/saga.ts:303]: nit: we could only wait for the receipt of the TX with the highest nonce (the last one).

**Review by jeanregisser — APPROVED** [review]
Awesome! 🚀 

I wonder if we should start to allow calling sagas from components directly like we did for the hackathon.
This would alleviate the need for intermediary Redux state and allow lifting the restriction of having 1 at a time. 🤔 
https://github.com/valora-labs/mini-apps-hack-wallet/blob/5cfd4e0f6a833e5a1b2e4bcdb705f6387b93bbcc/src/miniApps/earn/StakeScreen.tsx#L144

**Review by bakoushin — APPROVED** [review]
💰🔗💰

File: src/jumpstart/saga.ts
Comment (kathaypacific) [reply src/jumpstart/saga.ts:227]: i had that before, and removed it - https://github.com/valora-inc/wallet/pull/5091#discussion_r1524621485 essentially i don't think that it's necessary to be so overly defensive in this saga because we are creating the transactions ourselves. if we do want to decode the transactions, we should do it both for the approval and the actual transaction in order to populate the fields for the pending transactions. however then we'd also need to handle errors / decide what to do when the transactions are _not_ what we expect, which seems like a lot of extra unnecessary thinking if the transactions were created by us in the first place....

Comment (kathaypacific) [reply src/jumpstart/saga.ts:227]: happy to discuss this further, i can be persuaded to bring back the transaction decoding but i'll keep it simple and merge as is for now to unblock the other PRs relating to this

**Pull request merged**
